main/src/ca/uwaterloo/flix/language/phase/Namer.scala

```diff
@@ -1705,11 +1720,22 @@ object Namer extends Phase[WeededAst.Program, NamedAst.Root] {
   /**
     * Merges the given `uses` into the given use environment `uenv0`.
     */
-  private def mergeUseEnvs(uses: List[WeededAst.Use], uenv0: UseEnv): Validation[UseEnv, NameError] =
+  private def mergeUseEnvs(uses: List[WeededAst.Use], uenv0: UseEnv): Validation[UseEnv, NameError] = {
+
+    def locationOfUseDefOrSig(name: String, uenv: UseEnv): Option[SourceLocation] = {
```

Comment:
Add comment.

I understand why you are doing this: You want to know if there already a definition or signature, and you return a Location because def/sigs do not share anything else. Still it feels a big ugly.

As for the impl. itself I think a use of a map/flatMap could make it more pretty. But overall the design is a bit iffy.

Not sure exactly how to clean it up. We could introduce a small local ADT:

```
NotDefined
OtherDef(sym)
OtherSig(sym)
```

Reply:
I mean, maybe its fine as it is...